velox/common/caching/tests/SsdFileTest.cpp

```diff
@@ -54,9 +58,18 @@ class SsdFileTest : public testing::Test {
       int64_t maxBytes,
       int64_t ssdBytes = 0,
       bool setNoCowFlag = false) {
+    if (cache_ != nullptr) {
+      cache_->shutdown();
+    }
+    cache_.reset();
+    allocator_.reset();
+
     // tmpfs does not support O_DIRECT, so turn this off for testing.
     FLAGS_ssd_odirect = false;
-    cache_ = AsyncDataCache::create(MemoryAllocator::getInstance());
```

Comment:
Why we can't reply on MemoryAllocator::getInstance() returns allocator? Thanks!

Reply:
@xiaoxmeng MemoryAllocator::getInstance() sees the static member `customInstance_` is not null and directly returns it without downcasting to the subclass objects, then the virtual function calls on the returned MemoryAllocator object failed. You can check details in the Github issue https://github.com/facebookincubator/velox/issues/8188. 

I'm not sure if the incomplete MemoryAllocator object was due to incomplete destruction of the allocator_ object in previous tests, or it simply needs to be downcast in MemoryAllocator::getInstance(). Nonetheless, I think SsdFileTest should create a real allocator and that's why I made this fix. No matter if `MemoryAllocator::getInstance()` shall be fixed or not, this commit doesn't hurt and it unblocks me from the CI failure in the Iceberg PRs. But maybe what I thought was not entirely correct, and it would be great if you could find out more why it happened. Thanks!